core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -2219,6 +2207,42 @@ protected void convertFrom(
     }
   }
 
+  private List<String> getFieldAliases(SqlCall asCall) {
```

Comment:
This method needs javadoc

Is returning null the best thing? If it's illegal to call the method, you should throw. I hate nullable returns.

Conformance usually applies during validation, not during sql-to-rel. You know validation has succeeded at this point. If only one thing is possible, you should just do it.

Reply:
This method will be deleted in @danny0405 patch. So I will resolve this comment.